File: daily/230-kth-smallest-element-in-a-bst.py
class Solution:
    def kthSmallest(self, root: Optional[TreeNode], k: int) -> int:
        results = []
        def dfs(node: Optional[TreeNode]):
            if not node:
                return
            dfs(node.left)
            results.append(node.val)
            dfs(node.right)
        dfs(root)
        return results[k - 1]

File: daily/230-kth-smallest-element-in-a-bst.go
package daily

func kthSmallestTree(root *TreeNode, k int) int {
	var results []int
	dfskthSmallestTree(root, &results)
	return results[k-1]
}

func dfskthSmallestTree(node *TreeNode, results *[]int) {
	if node == nil {
		return
	}
	dfskthSmallestTree(node.Left, results)
	*results = append(*results, node.Val)
	dfskthSmallestTree(node.Right, results)
}
